lib/base/heimbase.c

```diff
@@ -86,11 +86,13 @@ struct heim_auto_release {
 void *
 heim_retain(void *ptr)
 {
-    struct heim_base *p = PTR2BASE(ptr);
+    struct heim_base *p = NULL;
```

Comment:
A compiler that blows up because of undefined pointer arithmetic (which, naturally, doesn't involve division by zero) where we never deref that pointer or act on the arithmetic is a pretty evil compiler...